core/src/main/java/org/mule/api/registry/LifecycleRegistry.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.api.registry;
+
+import org.mule.api.MuleException;
+import org.mule.api.lifecycle.Injector;
+
+/**
+ * A {@link Registry} which not only registers and fetches objects,
+ * but it's also capable of applying lifecycle and injects dependencies
+ *
+ * @since 3.7.0
+ */
+public interface LifecycleRegistry extends Registry, Injector
```

Comment:
Registry shoudn't implement Injector and expose a injectObject() method.  Rather concerns should be seperated and injector should be a seperate concept where a injector implementation uses a registry to inject. 

This was discussed with @pablokraan and @marianogonzalez and should be changed for this PR.